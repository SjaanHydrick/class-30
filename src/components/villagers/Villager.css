@font-face {
    font-family: 'AniCro';
    src: url('../../../src/fonts/Brightly\ Crush\ Shine.otf');
}

body, html {
    margin: 0;
    background-color: aliceblue;
}

.villagerList {
    display: flex;
    flex-flow: row wrap;
    justify-content: space-evenly;
    align-items: center;
    text-align: center;
}

a {
    text-decoration: none;
}

figure {
    padding-top: 25px;
    width: 200px;
    height: 300px;
    border: solid 5px seagreen;
    border-radius: 15px;
    background-color: white;
}

figure:hover {
    background-color: lightyellow;
    transition-duration: 1s;
}

.name {
    font-family: AniCro;
    font-size: 20pt;
    color: darkslategrey;
}

img {
    height: 200px;
}

.details {
    text-align: center;
    font-family: Georgia;
}

.back {
    text-align: left;
    font-family: AniCro;
    font-size: 15pt;
    color: darkslategrey;
}

.detailsImg {
    margin-top: 2%;
    height: 400px;
    margin-bottom: 2%;
}

.detailsName {
    font-family: AniCro;
    font-size: 30pt;
    margin-bottom: 1%;
    color: seagreen;
}

.quote {
    font-style: italic;
}