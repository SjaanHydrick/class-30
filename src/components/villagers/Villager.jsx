import React from 'react';
import PropTypes from 'prop-types';

const Villager = ({ image, name }) => (
  <figure>
    {
      !image || image === undefined ?
        <p>No Image Found</p>
        :
        <img src={image} />
    }
    <p>{name}</p>
  </figure>
);

Villager.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string.isRequired
};

export default Villager;
