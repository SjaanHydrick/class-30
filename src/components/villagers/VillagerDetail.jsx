/* eslint-disable max-len */
import React, { useState, useEffect } from 'react';
import { getVillager } from '../services/villagerApi';
import { Link, useParams } from 'react-router-dom';

const VillagerDetail = () => {
  const [chosenVillager, setVillager] = useState();
  const { id } = useParams();

  useEffect(async() => {
    const villager = await getVillager(id);
    setVillager(villager);
  }, []);

  return (
    <>
      {chosenVillager ?
        <div>
          <Link to="/"><p>&larr; Back to Home</p></Link>
          <img src={chosenVillager.image} />
          <p>{chosenVillager.name}</p>
          <p>&quot;{chosenVillager.quote}&quot;</p>
          <p>Personality: {chosenVillager.personality}</p>
          <p>Species: {chosenVillager.species}</p>
          <p>Birthday: {chosenVillager.birthday}</p>
          <p>Coffee: {chosenVillager.coffee.roast}, {chosenVillager.coffee.milk}, {chosenVillager.coffee.sugar}</p>
        </div>  
        :
        <p>Loading...</p>  
      }
    </>
  );
};

export default VillagerDetail;
