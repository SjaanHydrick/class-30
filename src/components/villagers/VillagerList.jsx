import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Villager from './Villager';
import styles from './Villager.css';

const VillagerList = ({ villagers }) => {
  const villagerElements = villagers.map((villager, index) => (
    <div key={index}>
      <Link to={`/${villager._id}`}>
        <Villager {...villager} />
      </Link>
    </div>
  ));

  return (
    <div data-testid="villagers" className={styles.villagerList}>
      {villagerElements}
    </div>
  );
};

VillagerList.propTypes = {
  villagers: PropTypes.arrayOf(PropTypes.shape({
    _id: PropTypes.string,
    name: PropTypes.string,
    image: PropTypes.string
  }))
};

export default VillagerList;
