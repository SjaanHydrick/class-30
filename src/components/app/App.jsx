import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Header from '../header/Header';
import AllVillagers from '../../containers/AllVillagers';
import VillagerDetail from '../../components/villagers/VillagerDetail';

export default function App() {
  return (
    <div>
      <Header />
      <Router>
        <Switch>
          <Route exact path="/" component={AllVillagers} />
          <Route path="/:id" component={VillagerDetail} />
        </Switch>
      </Router>
    </div>
  );
}
