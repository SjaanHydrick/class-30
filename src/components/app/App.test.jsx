/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from './App';
import Villager from '../villagers/Villager';
import VillagerDetail from '../villagers/VillagerDetail';
import VillagerList from '../villagers/VillagerList';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders Villagers', () => {
    const villagers = [{ name: '', image: '' }];
    const { asFragment } = render(
      <BrowserRouter>
        <VillagerList villagers={villagers}/>
      </BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders Villager', () => {
    const villager = [{
      '_id': '5f5fb4bbbfd05c2aed82e52c',
      'url': 'https://animalcrossing.fandom.com/wiki/Kyle',
      'name': 'Kyle',
      'japaneseName': 'リカルド Rikarudo',
      'image': 'https://vignette.wikia.nocookie.net/animalcrossing/images/4/46/Kyle_NH.png/revision/latest?cb=20200821045757',
      'quote': 'Never trust a sheep in wolf\'s clothing.',
      'gender': 'male',
      'personality': 'smug',
      'species': 'wolf',
      'birthday': 'December 6th (Sagittarius)',
      'phrase': 'alpha',
      'skill': 'Typing',
      'goal': 'Singer',
      'coffee': {
        'roast': 'Blue Mountain',
        'milk': 'Regular amount of milk',
        'sugar': 'Two spoonfuls of sugar'
      },
      'song': {
        'name': 'K.K. Metal',
        'link': 'https://animalcrossing.fandom.com/wiki/K.K._Metal'
      },
      'style': 'Rock \'n\' roll',
      '__v': 0
    }];

    const { asFragment } = render(
      <BrowserRouter>
        <Villager {...villager}/>
      </BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders VillagerDetail', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <VillagerDetail path={'/5f5fb4bbbfd05c2aed82e460'}/>
      </BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
