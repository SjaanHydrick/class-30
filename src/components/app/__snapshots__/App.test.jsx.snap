// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <h1>
      AC VILLAGERS
    </h1>
  </div>
  <div
    data-testid="villagers"
  >
    <div>
      <div
        class="villagerList"
      />
    </div>
  </div>
</DocumentFragment>
`;
